aws/validators.go

```diff
@@ -2407,3 +2407,56 @@ func validateRoute53ResolverName(v interface{}, k string) (ws []string, errors [
 
 	return
 }
+
+func validateCloudWatchEventBusName(v interface{}, k string) (ws []string, errors []error) {
+	// The names of custom event buses can't contain the / character. You can't
+	// use the name default for a custom event bus because this name is already
+	// used for your account's default event bus.
+	//
+	// If this is a partner event bus, the name must exactly match the name of the
+	// partner event source that this event bus is matched to. This name will include
+	// the / character.
+	value := v.(string)
+
+	if strings.HasPrefix(value, "aws.") {
+		// should be a partner event bus
+		return validateCloudWatchEventSourceName(v, k)
+	}
+
+	if len(value) < 1 {
+		errors = append(errors, fmt.Errorf("%q cannot be less than 1 character: %q", k, value))
+	} else if len(value) > 256 {
+		errors = append(errors, fmt.Errorf("%q cannot be longer than 256 characters: %q", k, value))
+	} else if value == "default" {
```

Comment:
I tried this PR and used it to specify the `event_bus_name` attribute on `aws_cloudwatch_event_rule` and `aws_cloudwatch_event_target` but this check for `default` leads to an issue. 
The `event_bus_name` `default` does not need to be specified but when you omit this it will be added to the statefile because the AWS API contains it in the response. This will then lead to a change on the next apply as it would be changed from `default` to `null`.
As this validation forbids to specify `default` it currently is not working.

I suggest to have different validations for the `event_bus_name` attribute on `aws_cloudwatch_event_rule` and `aws_cloudwatch_event_target`  and for the `name` attribute on the `aws_cloudwatch_event_bus`

Reply:
Good catch! I would leave the current validation though. The issue was that `event_bus_name` in the statefile was updated with a value from AWS API. I think `event_bus_name` is kinda part of the identification for `aws_cloudwatch_event_rule` and `aws_cloudwatch_event_target`, so it shouldn't be updated. At least that's how I fixed now. Otherwise I think that `event_bus_name` should be a required attribute. But I prefer as an optional attribute to not break current configurations. The idea is that we should always omit `event_bus_name` if we want to use the `default` event bus.
Also I changed the import for `aws_cloudwatch_event_rule` and `aws_cloudwatch_event_target`. I use `#` instead of `/`, eg `event-bus-name#rule-name#target-id` or `rule-name#target-id` for the default event bus, because an event bus name can contain `/`, it's possible to detect such names, since all of them start with `aws.partner/`. Not sure, if it's better to use `/` or `#` for imports here 🤔

Reply:
I reverted the some changes, so `#` is not used to import `aws_cloudwatch_event_rule` and `aws_cloudwatch_event_target`